--- 
title: |
  Extended Figure 4 \nA deeper dive into relationship between ACOA pathway abundance and microbiome taxonomy.
format:
  html:
    css: styles.css
---



```{r}
#| echo: false
#| warning: false
#| include: false
#| label: library_data_imports

library(tidyverse)
library(ggplot2)
library(phyloseq)
library(microViz)
library(survival)
library(survminer)
library(viridis)
library(yingtools2)
source("utility_funcs.R")


```

Some of these cells will be executable using only the public data.  These section panels <span class="executable" > Look like this</span>.  Other sections rely on data we are not able to make public (for example patient-level survival data).  These section panels <span class="nogo" > Look like this</span>.


```{r}
#| warning: false
#| echo: false
#| label: load_and_prep_data

plotting_ps <- readRDS("../data/cleaned_data/plotting_phyloseq_deidentified.rds") %>%
  ps_melt() %>% 
  mutate(acoa = exp(log_acoa)) %>%
  group_by(Sample, Family, acoa) %>%
  summarize(Abundance = sum(Abundance)) %>%
  ungroup()

```

## Relationfhip between relative abundance of top families and Acetyl-CoA pathway abundance
<h2 class="executable" >Panel A</h2>

Relationship between Rel Abundance of Lachnospiraceae and Acetyle-CoA pathway abundance. 

```{r}
#| warning: false
#| label: panel_a_Lachnospiraceae


plotting_ps %>%
  filter(grepl("Lachnospiraceae", Family)) %>%
  ggplot() +
  geom_smooth(aes(x=acoa, y=Abundance), method=lm, color="steelblue") +
  geom_point(aes(x=acoa, y=Abundance),
             shape=21, size=3, fill="steelblue", alpha=0.8) +
  theme_classic() + 
  labs(
    x = "Acetyl-CoA pathway gene abundance (RPKM)",
    y = "Lachnospiraceae"
  )

ggsave("svg/extended_figure_4_part_a_Lachnospiraceae.svg", w=5, h = 5)
```

```{r}
#|warning: false
#|label: p_value_panel_a
fit <- lm(Abundance ~ acoa, plotting_ps %>%
  filter(grepl("Lachnospiraceae", Family)))
summary(fit)
```

<h2 class="executable" >Panel B</h2>

Relationship between Rel Abundance of Ruminococcacae and Acetyle-CoA pathway abundance. 

```{r}
#| warning: false
#| label: panel_b_ruminococcaceae


plotting_ps %>%
  filter(grepl("Ruminococcaceae", Family)) %>%
  ggplot() +
  geom_smooth(aes(x=acoa, y=Abundance), method=lm, color="goldenrod") +
  geom_point(aes(x=acoa, y=Abundance),
             shape=21, size=3, fill="goldenrod", alpha=0.8) +
  theme_classic() + 
  labs(
    x = "Acetyl-CoA pathway gene abundance (RPKM)",
    y = "Ruminococcaceae"
  )

ggsave("svg/extended_figure_4_part_b_Ruminococcaceae.svg", w=5, h = 5)

```

```{r}
#|warning: false
#|label: p_value_panel_b
fit <- lm(Abundance ~ acoa, plotting_ps %>%
  filter(grepl("Ruminococcaceae", Family)))
summary(fit)
```



<h2 class="executable" >Panel C</h2>

Relationship between Rel Abundance of Ruminococcacae and Acetyle-CoA pathway abundance. 

```{r}
#| warning: false
#| label: panel_c_bacteroidacea


plotting_ps %>%
  filter(grepl("Bacteroidaceae", Family)) %>%
  ggplot() +
  geom_smooth(aes(x=acoa, y=Abundance), method=lm, color="firebrick2") +
  geom_point(aes(x=acoa, y=Abundance),
             shape=21, size=3, fill="firebrick2", alpha=0.8) +
  theme_classic() + 
  labs(
    x = "Acetyl-CoA pathway gene abundance (RPKM)",
    y = "Bacteroidaceae"
  )

ggsave("svg/extended_figure_4_part_c_Bacteroidaceae.svg", w=5, h = 5)
```

```{r}
#|warning: false
#|label: p_value_panel_c
fit <- lm(Abundance ~ acoa, plotting_ps %>%
  filter(grepl("Bacteroidaceae", Family)))
summary(fit)
```
