--- 
title: |
  <img src="./images/fiber_beaker_image_food.jpg" width="250px" style="margin-left:5%; margin-right:5%"/>
  Figure 2 - Butyrate Production Capacity from ACoA Pathway and ICB outcome
---



```{r}
#| echo: false
#| warning: false
#| include: false
#| label: library_data_imports

library(tidyverse)
library(ggplot2)
library(phyloseq)
library(microViz)
library(survival)
library(survminer)
library(viridis)
library(yingtools2)
source("utility_funcs.R")


```

Here are the column names of the data we are plotting:

fouraminobutyrate, glutarate, lysine and pyruvate represent different parts of the butyrate synthesis pathway. 

```{r}
#| warning: false
#| echo: false
#| label: load_data

acoa_df <- readRDS("../data/cleaned_data/acoa_data.rds")
colnames(acoa_df)
```

## Survival plots as a function of pyruvate gene abundance:

### Panel A

Overall Survival (OS) by above/below median ACoA pathway gene abundance. 

```{r}
#| warning: false
#| label: panel_a_os

# cut the distribution of pyruvate into above/below median:
acoa_df <- acoa_df %>%
  filter(!is.na(pyruvate)) %>%
  mutate(med_p = cut(pyruvate, quantile(pyruvate, probs=seq(0,1,0.5)), c("Below Median", "Above Median"))) %>% #ifelse(pyruvate > median(pyruvate, na.rm = T), "Above Median", "Below Median")) %>%
  mutate(tt_os_months = tt_os_d/30.44) %>%
  mutate(tt_pfs_months = tt_pfs_d/30.44) %>%
  arrange(med_p)

fit<- survfit(Surv(tt_os_months, event_os) ~ med_p, data = acoa_df)
names(fit$strata) <- gsub("med_p=", "", names(fit$strata))
p <- ggsurvplot(fit,  size = 1,  # change line size
           linetype = "solid", # use solid line
           break.time.by = 12, # break time axis by 12 months
           palette = c("aquamarine4", "coral4"), # custom color palette
           conf.int = TRUE, # Add confidence interval
           pval = TRUE, # Add p-value,
           legend.title="",
           risk.table = TRUE, risk.table.y.text.col = TRUE,
           ylab = "Survival %",
           xlab = "Time (Months)",
           xlim = c(0,48)
           ) 
print(p)

```
Same plot but looking at PFS instead of OS:

```{r}
#| warning: false
#| label: panel_a_pfs


fit<- survfit(Surv(tt_pfs_months, pfs_event) ~ med_p, data = acoa_df)
names(fit$strata) <- gsub("med_p=", "", names(fit$strata))
p <- ggsurvplot(fit,  size = 1,  # change line size
           linetype = "solid", # use solid line
           break.time.by = 12, # break time axis by 12 months
           palette = c("coral4", "aquamarine4"), # custom color palette
           conf.int = TRUE, # Add confidence interval
           pval = TRUE, # Add p-value,
           legend.title="",
           risk.table = TRUE, risk.table.y.text.col = TRUE,
           ylab = "Survival %",
           xlab = "Time (Months)",
           xlim = c(0,48),
           ggtheme = theme_classic() + theme(aspect.ratio = 1),
           tables.theme =  theme(aspect.ratio = 0.2)
           ) 
print(p)
```

### Panel B

Restricted cubic spline, adjusted for age, diagnosis and performance with above/below median pathway abundance. 

```{r}
#| warning: false
#| label: panel_b

model_spline_os <- coxph(Surv(tt_os_d, event_os) ~ rms::rcs(pyruvate,4) + age10 + factor(cohort) + ecog, acoa_df)
ptemp <- termplot(model_spline_os, se=T, plot=F)
buterm <- ptemp$pyruvate
center <- buterm$y[which(buterm$y == median(buterm$y))]
ytemp <- buterm$y + outer(buterm$se, c(0, -1.96, 1.96), '*')
exp_ytemp <- exp(ytemp - center)

spline_data <- data.frame(buty = buterm$x, Estimate = exp_ytemp[,1],
                          Lower = exp_ytemp[,2], Upper = exp_ytemp[,3])

ggplot(spline_data, aes(x = buty)) +
  geom_ribbon(aes(ymin = Lower, ymax = Upper), fill = "grey80", alpha = 0.5) +
  geom_line(aes(y = Estimate), color = "blue") +
  geom_rug(sides = "b") +  # Add rug plot at the bottom ('b') of the plot
  scale_y_log10() +  # Log scale for y-axis
  labs(x = "Pyruvate-pathway genes present", y = "Hazard ratio") +
  theme_minimal()

```

## Taxonomy breakdown plots:

We will now plot some microbiome data, using a `phyloseq` object that contains our sequencing data.  This object contains a list of the taxonomy of organisms identified, the abundance in each sample, and any associated sample specific metadata in one object. 

### Panel C

PCoA shows relationship between ACoA pathways gene abundance and PCA axes. 

```{r}
#| warning: false
#| label: panel_c

ps <- readRDS("../data/cleaned_data/plotting_phyloseq.rds")

ps %>%
  ps_filter(SAMPID %in% acoa_df$SAMPID) %>%
  tax_fix() %>%
  tax_transform(trans = "compositional", rank = "Family") %>%
  ord_calc("PCA") %>%
  ord_plot(fill = "log_acoa", size = 4, shape=21, alpha=0.6,
           plot_taxa = c("Lachnospiraceae", "Bacteroidales_order", "Ruminococcaceae")) +
  scale_fill_viridis(option="B", begin = 0, end=1) +
  theme_classic() + 
  xlim(c(-.25, .75)) +
  ylim(c(-.3, .6))

```

## Panel D

ACoA pathway abundance vs relative abundance of Lachnospiraceae and Reuminococcaceae 

```{r}
#| warning: false
#| label: panel_d_lachno

# We will take the information we need out the phyloseq, and convert it to a dataframe for plotting: 
compare_family <- ps %>% 
  get.otu.melt() %>%
  group_by(SAMPID, Family) %>%
  summarise(total_family = sum(pctseqs)) %>%
  left_join(acoa_df %>% select(SAMPID, `pyruvate`))

compare_family %>%
  filter(Family == "Lachnospiraceae") %>%
  ggplot() +
  geom_smooth(aes(x=pyruvate, y=total_family), method=lm, color="steelblue") +
  geom_point(aes(x=pyruvate, y=total_family),
             shape=21, size=3, fill="steelblue", alpha=0.8) +
  theme_classic() + 
  labs(
    x = "Acetyl-CoA pathway gene abundance (RPKM)",
    y = "Lachnospiraceae"
  )

```
```{r}
#| warning: false
#| label: panel_d_rumino


compare_family %>%
  filter(Family == "Ruminococcaceae") %>%
  ggplot() +
  geom_smooth(aes(x=pyruvate, y=total_family), method=lm, color="darkgoldenrod3") +
  geom_point(aes(x=pyruvate, y=total_family),
             shape=21, size=3, fill="darkgoldenrod3", alpha=0.8) +
  theme_classic() + 
  labs(
    x = "Acetyl-CoA pathway gene abundance (RPKM)",
    y = "Lachnospiraceae"
  )

```
## Panel E
Univariate Cox regression analyses PFS relationship to ACoA gene abundance in external cohort.

```{r}
#| warning: false
#| label: panel_e


```

